### ISSUE TITLE
Async response overrides the original 202

### ISSUE THREAD

**Issue by devcritter** [jembi/openhim-mediator-engine-java#3]
See discussion in PR #2 

During async processing, the mediator will respond to core with a status of 202 (Accepted) and then later update the transaction with the final response via the API. However this update overrides the original 202 route with the final response.

Should we think of a way to keep the 202 response, and somehow add the final response, perhaps to another route?

cc @rcrichton @armageddon @bausmeier

**Comment by rcrichton** [comment]
Do we capture the primary route details with the rest of the routes in the transactions object. I think we do. If so maybe what we could do is leave the `transaction.response` as the 202 but update the route response (`transaction.routes[0].response`) with the detail that we received asynchronously. That way both responses are captured but it clearly shows that the transaction was handled asynchronously and that the client received a 202.

**Comment by devcritter** [comment]
Actually I'm not certain we do: https://github.com/jembi/openhim-core-js/blob/master/test/unit/messageStoreTest.coffee#L154-L172. It's looks like `routes` is only being added to on non-primary. But so maybe we should add handling to add the primary to the routes object as well?

Hmmm, also what if the async mediator is being called from a non-primary route?

Maybe we could have the async mediator _add_ a route to the transaction when it finishes...

**Comment by rcrichton** [comment]
Yeah I think it makes sense to add the primary response to `routes` object in all cases. If this happens do we need to still add a route to the transaction when it finishes? Or am I missing something?

**Comment by devcritter** [comment]
Yeah, I was thinking about the case where an async mediator is called from a non-primary route.
